checker/src/main/java/org/checkerframework/checker/resourceleak/ResourceLeakVisitor.java

```diff
@@ -80,6 +86,66 @@ public Void visitMethod(MethodTree node, Void p) {
     return super.visitMethod(node, p);
   }
 
+  // Overwritten to check that destructors (i.e. methods responsible for resolving
+  // the must-call obligations of owning fields) enforce a stronger version of
+  // @EnsuresCalledMethods: that the claimed @CalledMethods annotation is true on
+  // both exceptional and regular exits, not just on regular exits.
+  @Override
+  protected void checkPostcondition(
+      MethodTree methodTree, AnnotationMirror annotation, JavaExpression expression) {
+    super.checkPostcondition(methodTree, annotation, expression);
+    // Only check if the required annotation is a CalledMethods annotation (implying
+    // the method was annotated with @EnsuresCalledMethods).
+    if (!AnnotationUtils.areSameByName(
+        annotation, "org.checkerframework.checker.calledmethods.qual.CalledMethods")) {
+      return;
+    }
+    // This method might be a destructor that is responsible for resolving the must-call
+    // obligations of one or more owning fields of the containing class.
+    ExecutableElement elt = TreeUtils.elementFromDeclaration(methodTree);
+    TypeElement containingClass = ElementUtils.enclosingTypeElement(elt);
+    MustCallAnnotatedTypeFactory mustCallAnnotatedTypeFactory =
+        rlTypeFactory.getTypeFactoryOfSubchecker(MustCallChecker.class);
+    AnnotationMirror mcAnno =
+        mustCallAnnotatedTypeFactory
+            .getAnnotatedType(containingClass)
+            .getAnnotationInHierarchy(mustCallAnnotatedTypeFactory.TOP);
+    List<String> mcValues =
+        AnnotationUtils.getElementValueArray(
+            mcAnno, mustCallAnnotatedTypeFactory.getMustCallValueElement(), String.class);
+    String methodName = elt.getSimpleName().toString();
+    if (!mcValues.contains(methodName)) {
+      // Not a destructor, just a method with an ECM annotation. No further checking to do.
+      return;
+    }
```

Comment:
Can we extract this code to a new method that returns a boolean?  Something like `isDestructorForEnclosingClass`?